cisagov/manage.get.gov
# Issues: 1807,  2170:  User profile and header refactor [DK]

## Ticket

Resolves #1807 


## Changes


- 'Your profile' page when profile_feature flag is enabled
- 'Your profile' link in main nav when profile_feature flag is enabled
- 'Your contact information' removed from domain management pages when profile_feature flag is activated
- 'Your profile' not in main nav when profile_feature flag is disabled
- 'Your contact information' enabled in domain management pages when profile_feature is disabled
- Views for 'Your profile' and 'Your contact information' throw 403 error pages if they are attempted to be accessed directly

(Note that domain request workflow is not in scope for this ticket) 



## Context for reviewers



New feature, Your profile, replaces 'Your contact information' in domain management flow.  The appearance of the new feature is determined by the waffle flag, profile_feature, being enabled/disabled in django admin.

To test new feature:
Enable profile_feature by clicking profile_feature in the Waffle flags in django admin, and set Everyone to 'Yes'.
Navigate to app's home page.  You should see 'Your profile' in the nav.  Click the link to interact and update your profile.
Navigate to any domain.  You should see that 'Your contact information' is no longer in the domain detail section and is no longer in the left hand nav for domain management.

To test turning the new feature off:
Disable profile_feature by clicking profile_feature in the Waffle flags in django admin, and set Everyone to 'No'.
Navigate to app's home page.  You should NOT see 'Your profile' in the nav. 
Navigate to any domain.  You should see the 'Your contact information' in the domain detail section and in the left hand nav for domain management.

## Setup



## Code Review Verification Steps

### As the original developer, I have

#### Satisfied acceptance criteria and met development standards


#### Ensured code standards are met (Original Developer)


#### Validated user-facing changes (if applicable)


### As a code reviewer, I have

#### Reviewed, tested, and left feedback about the changes


#### Ensured code standards are met (Code reviewer)


#### Validated user-facing changes as a developer




**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

### As a designer reviewer, I have

#### Verified that the changes match the design intention


#### Validated user-facing changes as a designer




## Screenshots

Main Nav:
<img width="1073" alt="Screenshot 2024-05-16 at 3 52 49 PM" src="https://github.com/cisagov/manage.get.gov/assets/107004823/d28b055b-dfdd-43c1-ab3f-061a70a29da4">

<img width="803" alt="Screenshot 2024-05-16 at 3 52 59 PM" src="https://github.com/cisagov/manage.get.gov/assets/107004823/7809d20b-1c03-4bf3-a970-6bf92bea9785">

Profile page:
<img width="1008" alt="Screenshot 2024-05-16 at 3 54 31 PM" src="https://github.com/cisagov/manage.get.gov/assets/107004823/d95b5687-f998-4678-b335-f86d61c420bb">
